src/backend/db/queries.js

```diff
@@ -1,389 +1,382 @@
-const knex = require('./knex.js');
+const knex = require("./knex.js");
 
 //Users
 
-const addUser = async ({name, pwd}) => {
-  return knex('Users')
-         .insert({name: name, pwd: pwd})
-         .returning('name')
-}
+const addUser = async ({ name, pwd }) => {
+  return knex("Users")
+    .insert({ name: name, pwd: pwd })
+    .returning("name");
+};
 
 const getUsers = () => {
-    return knex('Users').select();
-}
-
-const findUser = ({name}) => {
-    return knex('Users').where({name: name}).select();
-}
+  return knex("Users").select();
+};
 
+const findUser = ({ name }) => {
+  return knex("Users")
+    .where({ name: name })
+    .select();
```

Comment:
here's where i would do something like
```js
const user = knex("users").where('name', name).first();
if (!user) {
  throw new NotFound(`User ${name} does not exist`);
}

return user;
```

where `NotFound` is an error class that your error middleware knows how to handle. something like

```js
class NotFound extends Error {
  status = 404;
}
```

would mean that your error handler knows to set the response status to 404 if `NotFound` is thrown.

Reply:
Knex returns a promise that needs to be resolved later. Do you suggest resolving this promise here and returning the result like the below snippet?
```
const user = await knex("users").where('name', name).first();
if (!user) {
  throw new NotFound(`User ${name} does not exist`);
}

return user;
```

Also if I throw this error here will i need to handle the error in the route? Or is there a way to forward the error to the handler in app.js?

Reply:
yep that would be fine. the consumer of this function is already resolving the returned promise because the function is `async` so this just adds the possibility that the promise is rejected because the user doesn't exist.

there's no need to forward the error to the handler since you have error handling middleware. take a look at my earlier comment about fixing your 404 and error handlers - once that is fixed then you should get a `404 User foo does not exist` response without adding any extra error handling here

Reply:
Thank you! You made me rethink how to handle errors in Express! I now think I have an elegant solution which properly uses middleware to handle errors. I'll make another PR if you want to take a look but no problem if you don't have time!